.github/workflows/cli.yml

```diff
@@ -39,47 +47,578 @@ jobs:
+      - name: Discover test cases and set outputs
+        id: discover-tests
         run: |
           cd azure-cli-extensions
           source .venv/bin/activate
-          azdev test apic-extension --live
-        env:
-          USERASSIGNED_IDENTITY: ${{ secrets.USERASSIGNED_IDENTITY }}
-          AWS_ACCESS_KEY_LINK: ${{ secrets.AWS_ACCESS_KEY_LINK }}
-          AWS_SECRET_ACCESS_KEY_LINK: ${{ secrets.AWS_SECRET_ACCESS_KEY_LINK }}
+          cd src/apic-extension
 
+          # Check if specific test cases are provided
+          test_input='${{ inputs.test-cases }}'
+          if [ -n "$test_input" ] && [ "$test_input" != "null" ] && [ "$test_input" != "" ]; then
+            echo "Using specific test cases from input: $test_input"
+            test_cases="$test_input"
+          else
+            echo "Discovering all test cases using pytest..."
+
+            # Find test files and collect test cases
+            test_files=$(find . -name "test_*.py" -type f | head -10)
+            echo "Debug: Found test files:"
+            echo "$test_files"
+
+            if [ -z "$test_files" ]; then
+              echo "❌ No test files found"
+              test_cases='["full-suite"]'
+            else
+              # Use pytest to collect all test cases
+              echo "Debug: Collecting test cases with pytest..."
+
+              # Collect test node IDs and extract just the test function names
+              test_collection=$(python -m pytest --collect-only -q 2>/dev/null | \
+                grep "::test_" | \
+                sed 's/.*::\(test_[^[]*\).*/\1/' | \
+                sort | uniq || true)
+
+              echo "Debug: Raw test collection output:"
+              echo "$test_collection"
+
+              if [ -z "$test_collection" ]; then
+                echo "❌ No test cases found with pytest, trying alternative approach..."
+
+                # Fallback: Parse test files directly for test function definitions
+                test_collection=$(grep -h "^def test_" $(find . -name "test_*.py") | \
+                  sed 's/def \(test_[^(]*\).*/\1/' | \
+                  sort | uniq || true)
+
+                echo "Debug: Fallback test collection:"
+                echo "$test_collection"
+              fi
+
+              if [ -z "$test_collection" ]; then
+                echo "❌ No individual test cases found, using full-suite approach"
+                test_cases='["full-suite"]'
+              else
+                test_count=$(echo "$test_collection" | wc -l)
+                echo "✅ Successfully discovered $test_count test cases"
+
+                # Convert to JSON array
+                test_cases=$(echo "$test_collection" | jq -R -s -c 'split("\n") | map(select(length > 0))')
+                total_tests=$(echo "$test_cases" | jq '. | length')
+                echo "Found $total_tests test cases total"
+
+                # Validate JSON format
+                if ! echo "$test_cases" | jq empty 2>/dev/null; then
```

Comment:
Since `$test_cases` is generated in line 110, when will it be an invalid json string?